R/spaces.R

```diff
@@ -0,0 +1,137 @@
+spaces_base <- "nyc3.digitaloceanspaces.com"
+
+check_space_access <- function(spaces_key) {
+  tmp <- if (is.null(spaces_key)) Sys.getenv("DO_SPACES_ACCESS_KEY") else spaces_key
+  if (tmp == "") stop("Need a digital ocean spaces access key defined in your session", call. = FALSE) else tmp
+}
+
+check_space_secret <- function(spaces_secret) {
+  tmp <- if (is.null(spaces_secret)) Sys.getenv("DO_SPACES_SECRET_KEY") else spaces_secret
+  if (tmp == "") stop("Need a digital ocean spaces access key defined in your session", call. = FALSE) else tmp
+}
+
+#' @param x Object to coerce to a space
+#' @export
+#' @rdname spaces
+as.space <- function(x) UseMethod("as.space")
+#' @export
+as.space.space <- function(x) x
+
+#' @export
+print.space <- function(x, ...) {
+  cat("<space>", x$Name, "\n", sep = "")
+  cat("  Created at: ", x$CreationDate, "\n")
+}
+
+#' @export
+summary.space <- function(object, ...) {
+  space_info <- space_info(name = object$Name, ...)
+
+  # obtain total size used by space
+  size <- space_size(space_info)
+
+  # obtain number of files in space
+  n_files <- space_files(space_info)
+
+  cat("<space_detail>", object$Name, "\n", sep = "")
+  cat("  Size (GB):    ", size, "\n")
+  cat("  Files:        ", n_files, "\n", sep = "")
+  cat("  Created at:   ", object$CreationDate, "\n")
+}
+
+#' Spaces storage operations
+#'
+#' \describe{
+#'  \item{space}{List space contents}
```

Comment:
ah, maybe `space` == `space_info`

Reply:
I went back and forth with this, but now I'm going to omit a `space()` function since `spaces()` will grab everything and it is straight forward for the user to grab whichever space they want to work with.  They all have a special `space` class so we'll be able to add many more methods and operations as time goes on  I'll delete references to a `space` function in my next PR.

Reply:
Also I've made `space_info()` an internal function

Reply:
sounds good